tsl/src/bgw_policy/job.c

```diff
@@ -304,26 +303,29 @@ policy_retention_execute(int32 job_id, Jsonb *config)
 		PushActiveSnapshot(GetTransactionSnapshot());
 	}
 
-	table_relid = ts_hypertable_id_to_relid(policy_retention_get_hypertable_id(config));
-	hypertable = ts_hypertable_cache_get_cache_and_entry(table_relid, CACHE_FLAG_NONE, &hcache);
+	object_relid = ts_hypertable_id_to_relid(policy_retention_get_hypertable_id(config));
+	hypertable = ts_hypertable_cache_get_cache_and_entry(object_relid, CACHE_FLAG_NONE, &hcache);
 	open_dim = get_open_dimension_for_hypertable(hypertable);
 	boundary = get_window_boundary(open_dim,
 								   config,
 								   policy_retention_get_retention_window_int,
 								   policy_retention_get_retention_window_interval);
 	boundary_type = ts_dimension_get_partition_type(open_dim);
 
-	dc_temp = ts_chunk_do_drop_chunks(hypertable,
-									  boundary,
-									  InvalidOid,
-									  boundary_type,
-									  InvalidOid,
-									  DEBUG2,
-									  NULL);
-	num_dropped = list_length(dc_temp);
-	ts_cache_release(hcache);
+	/* We need to do a reverse lookup here since we might have the
+	   materialized hypertable and need to call drop_chunks with the
+	   continuous aggregate if that is the case. */
```

Comment:
It took me awhile to understand the comment. Here is my suggestion if it improves:
```suggestion
	/* We need to do a reverse lookup here since the given hypertable
	   might be a materialized hypertable, and thus need to call drop_chunks
	    on the continuous aggregate instead. */
```